website/src/app/blog/how-dns-based-traffic-routing-works/readme.mdx

```diff
@@ -0,0 +1,372 @@
+`github.com` might inadvertently route traffic for _other_ GitHub (and maybe
+even non-GitHub) services as well!
+
+#### NAT to the rescue?
+
+Ok, but what about another solution: can the organization simply configure its
+client devices to use the organization's DNS server to respond with internal IP
+addresses for the public services?
+
+That's much better -- it solves the problem of routing traffic for the wrong
+service, but it introduces a new problem: the organization now needs to maintain
+a list of all the public resources its workforce needs to access, then configure
+its DNS server to respond with a unique internal IP address for each one,
+finally it needs to be reasonably sure these IP addresses don't conflict with
+other networks available to the employee's machine, such as a home printer.
+
+<p className="text-sm text-center mx-auto italic">
+  Figure 6: Split DNS with NAT: A management nightmare
+</p>
+
+On top of the addressing headache, the organization needs to operate NAT routers
+that accept the traffic from the VPN clients across their workforce, translate
+the source addresses so they appear to come from one of the IPs in the
+allowlist, and finally forward the traffic on to the service in question. The IP
+addressing configuration and NAT router configuration would need to be kept in
+sync so that they function together.
+
+So we've taken what was originally a DNS problem and translated it to a
+configuration management problem. So let's solve it!
+
+## DNS-based traffic routing
+
+By now you're probably thinking, "this is all well and good, but what does this
+have to do with Firezone?". Glad you asked!
+
+Firezone's approach to DNS was designed to address both the insecurity of DNS
+and the shortcomings of Split DNS, into a single solution that's just as secure
+as Split DNS, but much more flexible and easier to manage.
+
+### How it works
+
+When a user signs in, the Firezone Client configures its host operating system
+to use a special, lightweight DNS proxy running inside the Client for all DNS
+queries on the system.
+
+This proxy lives at `100.100.111.0/24` for IPv4 queries and
+`fd00:2021:1111:8000:100:100:111:0/120` for IPv6 queries. These addresses are
+not routable on the public internet, so there's no risk that queries for
+internal resources will leave the device unexpectedly.
+
+The IP addresses the proxy chooses to "listen on" are determined by the
+[upstream resolvers](#configuring-client-dns-upstream-resolvers) available to
+the Client. For each IPv4 upstream resolver, the proxy listens on an address in
+the `100.100.111.0/24` range, and for each IPv6 upstream resolver, the proxy
+listens on an address in the `fd00:2021:1111:8000:100:100:111:0/120` range.
+
+<p className="text-sm text-center mx-auto italic">
+  Figure 7: Firezone's embedded stub resolver
+</p>
+
+When an application makes a DNS query, the proxy checks if the name matches a
```

Comment:
Saying that the proxy checks it is kind of not precise, connlib does the checking, we can say in the first paragraphthat "the client" intercept DNS requests and then forwards the queries to an embedded resolver(hickory!)

Reply:
I mention it's a stub resolver with "an interesting property" or something along those lines